src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,33 @@
+using form_builder.Models;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        {
+            var optionalifElement = page.Elements.Where(_ => !string.IsNullOrEmpty(_.Properties.OptionalIfValue) || !string.IsNullOrEmpty(_.Properties.OptionalIfNotValue));
+
+            if (optionalifElement.Any())
+            {
+                foreach (var e in optionalifElement)
+                {
+                    var userChoice = convertedAnswers.AllAnswers
+                        .Where(x => x.QuestionId == e.Properties.OptionalIf)
+                        .Select(x => x.Response)
```

Comment:
just use .FirstOrDefault here... don't need .Select and then pick FirstOrDefault